contrib/JLightning/src/de/renepickhardt/ln/UnixDomainSocketRpc.java

```diff
@@ -0,0 +1,119 @@
+package de.renepickhardt.ln;
+/**
+ * UnixDomainSocketRpc the base class to handle communication between
+ * JLightning and the c-lightning node over the UnixDomainSocket 
+ * 
+ * This file is basically a port of the pylightning python client library
+ * that comes with c-lightning.
+ * 
+ * The Author of this Java Client library is Rene Pickhardt. 
+ * He also holds the copyright of this file. The library is licensed with
+ * a BSD-style license. Have a look at the LICENSE file. 
+ * 
+ * If you like this library consider a donation via bitcoin or the lightning
+ * network at http://ln.rene-pickhardt.de
+ * 
+ * @author Rene Pickhardt
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.newsclub.net.unix.AFUNIXSocket;
+import org.newsclub.net.unix.AFUNIXSocketAddress;
+
+public class UnixDomainSocketRpc {
+	protected AFUNIXSocket sock;
+	protected InputStream is;
+	protected OutputStream os;
+	protected static int id = 1;
+	
+	public UnixDomainSocketRpc(String socket_path) {
+		File socketFile = new File(socket_path);
+		try {
+			this.sock = AFUNIXSocket.newInstance();
+			this.sock.connect(new AFUNIXSocketAddress(socketFile));
+			this.is = sock.getInputStream();
+			this.os = sock.getOutputStream();			
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	public String call(String method, HashMap<String, String> payload) {
+		// if no payload is given make empty one
+		if(payload == null) {
+			payload = new HashMap<String, String>();
+		}
+		
+		//remove null items from payload
```

Comment:
There's no need to add the keys to a new `HashSet` here. You can simply iterate through the `payload.keySet()` and [and remove an entry if value is `null`](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#remove-java.lang.Object-java.lang.Object-).

Here's some sample code:
```
Map<String, String> m = new HashMap<String, String>();
    m.put("1", "One");
    m.put("2", null);
    System.out.println(m);
    for(String k:m.keySet()) {
      m.remove(k, null);
    }
    System.out.println(m);
```

And output is:
```
{1=One, 2=null}
{1=One}
```